src/sentry_core.c

```diff
@@ -389,15 +390,17 @@ sentry__capture_event(sentry_value_t event)
                 SENTRY_WITH_OPTIONS_MUT (mut_options) {
                     sentry__envelope_add_session(
                         envelope, mut_options->session);
-                    // we're assuming that if a session is added to an envelope
-                    // it will be sent onwards.  This means we now need to set
-                    // the init flag to false because we're no longer the
-                    // initial session update.
+                    // we're assuming that if a session is added to an
+                    // envelope it will be sent onwards.  This means we now
+                    // need to set the init flag to false because we're no
+                    // longer the initial session update.
                     mut_options->session->init = false;
                 }
             }
 
             sentry__capture_envelope(options->transport, envelope);
+        } else {
+            was_captured = false;
```

Comment:
while updating events i noticed this had a subtle bug: `capture_event` returns the event ID even if the event fails to send due to sampling and/or other internal issues. this fix takes care of sampling, however `sentry__capture_envelope` can silently fail due to missing user consent. i'm not sure if this was a deliberate decision, ie we do not want to let SDK users know when user consent has not been provided or if this was just overlooked.

i've made note of this as a TODO in the tests, but any insight on this would be great.

Reply:
fixing this comment by re-attaching it to a different line, please hold